.buildkite/hooks/pre-command

```diff
@@ -62,13 +62,34 @@ if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats" || "$BUILDKITE_PIPELINE_SLUG" == "fi
   fi
 fi
 
-if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats-metricbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-libbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-packetbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-winlogbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-libbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-metricbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-packetbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-winlogbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-dockerlogbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-auditbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-filebeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-heartbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-osquerybeat" ]]; then
-  if [[ "${BUILDKITE_COMMAND}" =~ ^buildkite-agent ]]; then
-    echo "Skipped pre-command when running the Upload pipeline"
-    exit 0
+ENABLED_BEATS_PIPELINES_SLUGS=(
+  "auditbeat"
+  "filebeat"
+  "beats-metricbeat"
+  "beats-libbeat"
+  "beats-packetbeat"
+  "beats-winlogbeat"
+  "beats-winlogbeat"
+  "beats-xpack-libbeat"
+  "beats-xpack-metricbeat"
+  "beats-xpack-packetbeat"
+  "beats-xpack-winlogbeat"
+  "beats-xpack-dockerlogbeat"
+  "beats-xpack-auditbeat"
+  "beats-xpack-filebeat" 
+  "beats-xpack-heartbeat" 
+  "beats-xpack-osquerybeat"
+  )
+
+for slug in "${ENABLED_BEATS_PIPELINES_SLUGS[@]}"; do
+  if [[ "$BUILDKITE_PIPELINE_SLUG" == "$slug" ]]; then
+    if [[ "${BUILDKITE_COMMAND}" =~ ^buildkite-agent ]]; then
+      echo "Skipped pre-command when running the Upload pipeline"
+      exit 0
+    fi
+    source .buildkite/scripts/setenv.sh
```

Comment:
This is causing the pre-exit hook to fail, because we set the `BIN` inside the `setenv.sh` and we are not sourcing this sudently.

Reply:
Fixed